### PR TITLE
Refactor build_and_tests GHA workflow before for migrating publish jobs

### PR DESCRIPTION
**Pull request by shovnik**
## Which problem is this solving?

As part of [issue](https://github.com/open-telemetry/opentelemetry-collector/issues/1234), this pull request does some naming changes and refactoring on the already migrated build_and_test workflows jobs in preparation for migrating the remaining `windows-msi`, `publish-dev` and `publish-stable` jobs in the following [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/2299)
This PR must be reviewed and merged before merging the final [migration PR](https://github.com/open-telemetry/opentelemetry-collector/pull/2299). The purpose of this PR is to remove changes to existing jobs from the final migration PR.

## What changes are made in this PR?
- All step names are consistent across jobs (descriptive and capitalized except articles/prepositions)
- Bash logic that is complex or to be reused in the new jobs was extracted into shell scripts
- Permissions were added to binaries before upload to allows windows machines to execute them. This meant archiving the binaries with tar before upload for them to retain permissions.
cc- @alolita, @AzfaarQureshi

### CODE REVIEW

**Comment by shovnik** [comment]
@bogdandrutu I separated out the naming changes and refactoring to the existing workflow into a separate PR as requested. Please take a look.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -69,32 +69,18 @@ jobs:
...
-            fi
-          done
-          MATRIX+="]}"
-          echo "::set-output name=matrix::$MATRIX"
-      - name: Upload tool binaries
+        run: ./.github/workflows/scripts/setup_load_tests.sh
+      - name: Add Execute Permissions to Binaries
+        run: chmod -R +x /home/runner/go/bin
+      - name: Create Tool Binaries Archive
+        run: tar -cvf tool-bin.tar /home/runner/go/bin
```
Comment (bogdandrutu) [inline .github/workflows/build-and-test.yml:78]: why is this needed vs the whole path?

**Review by bogdandrutu — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (shovnik) [reply .github/workflows/build-and-test.yml:78]: The reason I archived the file was to preserve the execute permissions on the binaries created. Specifically on the windows-msi job, there was no way to easily `chmod +x` the binary once downloaded given the OS. Also, this prevents having to add permissions to the binaries everytime it is downloaded although extracting the archive is required instead.

Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:78]: I am fine to always add the chmod +x instead of archive. Maybe leave also a TODO to explain this.

Comment (shovnik) [reply .github/workflows/build-and-test.yml:78]: The only issue is that I cannot do the chmod for the windows job (In PR 3/3), this was a workaround specifically for that. I can remove the archiving from tool binaries, but it is needed for the collector binaries which the windows-msi job uses.

Comment (shovnik) [reply .github/workflows/build-and-test.yml:78]: Removed the archiving from all the tool binaries, because it was unnecessary. I still need to archive the collector binary produced from cross-compile because it requires execute permissions to run in `windows-msi`.

Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:78]: It does not seem to be reflected in the code.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -139,64 +121,63 @@ jobs:
...
           cache-name: cache-go-modules
         with:
           path: /home/runner/go/pkg/mod
           key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: Download Tool Binaries
+        uses: actions/download-artifact@v1
+        with:
+          name: tool-binaries
+      - name: Extract Tool Binaries Archive
+        run: tar -xvf tool-binaries/tool-bin.tar -C /
```
Comment (bogdandrutu) [inline .github/workflows/build-and-test.yml:180]: Why this?

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -285,30 +264,23 @@ jobs:
...
         with:
           name: collector-binaries
-          path: ./bin
-      - name: Add execute permissions to downloaded binaries
+      - name: Extract Binaries Archive
         run: |
-          chmod -R +x /home/runner/go/bin
-          chmod -R +x ./bin
+          tar -xvf tool-binaries/tool-bin.tar -C /
+          tar -xvf collector-binaries/bin.tar
```
Comment (bogdandrutu) [inline .github/workflows/build-and-test.yml:280]: Why this?

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (shovnik) [reply .github/workflows/build-and-test.yml]: I pushed my amend to the wrong Git branch, did not mean to mislead sorry. All instances of tool archive should be gone now.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (shovnik) [reply .github/workflows/build-and-test.yml]: Forgot to push removing this. Updated.

File: .github/workflows/build-and-test.yml (hunk 3 above)
Comment (shovnik) [reply .github/workflows/build-and-test.yml]: Forgot to push removing this. Updated.

**Review by bogdandrutu — APPROVED** [review]
